src/mixins/dialog.scss

```diff
@@ -92,26 +90,25 @@
 }
 
 @mixin dialog__wrapper {
-    display: table-cell;
-    height: 0;
-    overflow: hidden;
-    position: relative;
-    vertical-align: middle;
-    width: 0;
+    height: 100%;
 
-    .is-visible & {
-        height: 100%;
-        width: 100%;
+    @include respond-to-wide {
+        display: table-cell;
+        height: 100vh;
+        padding: 0 ($grid * 6);
+        text-align: center;
+        vertical-align: middle;
+        width: 100vw;
     }
 }
 
 @mixin dialog__overlay {
     background: transparentize(map-get($colors, black), .2);
     bottom: 0;
-    display: table;
     height: 100%;
     left: 0;
     opacity: 0;
+    overflow-y: auto;
```

Comment:
I'm on my phone right now so can't check, but you might want to add the touch-scroll thingy to get smoother scrolling on mobile https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-overflow-scrolling